fasthtml/core.py

```diff
@@ -459,6 +459,9 @@ def __init__(self, debug=False, routes=None, middleware=None, exception_handlers
             if htmx: hdrs = [htmxscr] + hdrs
             hdrs = [charset, viewport] + hdrs
         excs = {k:_wrap_ex(v, hdrs, ftrs, htmlkw, bodykw) for k,v in (exception_handlers or {}).items()}
+        if 404 not in excs: 
```

Comment:
Nice! To avoid the dupe code here, move it above the previous line, and do `if exception_handlers and 404 not in exception_handlers` or something like that. Then you won't have to have a separate `_wrap_ex` for it.